src/interfaces/token.ts

```diff
@@ -64,20 +64,21 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
       .balances(address)
       .then(balances => balances.filter(token => token.balance !== "0"));
 
-    if (this.chainId === 1 || this.chainId === 1337) {
-      let zapperBalances = await this.yearn.services.zapper.balances(address);
-      const vaultBalanceAddresses = new Set(vaultBalances.map(balance => balance.address));
-      zapperBalances = zapperBalances.filter(balance => !vaultBalanceAddresses.has(balance.address));
-      return zapperBalances.concat(vaultBalances);
-    }
-
-    if (this.chainId === 250 || this.chainId === 42161) {
-      let ironBankTokens = await this.yearn.ironBank.balances(address);
-      const vaultBalanceAddresses = new Set(vaultBalances.map(balance => balance.address));
-      ironBankTokens = ironBankTokens.filter(balance => !vaultBalanceAddresses.has(balance.address));
-      return ironBankTokens.concat(vaultBalances);
+    switch (this.chainId) {
```

Comment:
I would move the throw to the default case

```typescript
    switch (this.chainId) {
      case 1 || 1337: {
        let zapperBalances = await this.yearn.services.zapper.balances(address);
        const vaultBalanceAddresses = new Set(
          vaultBalances.map((balance) => balance.address)
        );
        zapperBalances = zapperBalances.filter(
          (balance) => !vaultBalanceAddresses.has(balance.address)
        );
        return zapperBalances.concat(vaultBalances);
      }
      case 250 || 42161: {
        let ironBankTokens = await this.yearn.ironBank.balances(address);
        const vaultBalanceAddresses = new Set(
          vaultBalances.map((balance) => balance.address)
        );
        ironBankTokens = ironBankTokens.filter(
          (balance) => !vaultBalanceAddresses.has(balance.address)
        );
        return ironBankTokens.concat(vaultBalances);
      }
      default:
        throw new SdkError(`the chain ${this.chainId} hasn't been implemented yet`);
    }
```

Reply:
Hello, Does this mean that the PR is updated and waiting to be merged, or do I need to edit any code/commit from my end? I'm new to open source, thanks for the help!

Reply:
Hi Mahmoud, I've given you a suggestion, so now it's up to you to either change your code to reflect my suggestion or to give your reasoning for not doing the above. We've decided to go with `console.erro` instead of throwing with https://github.com/yearn/yearn-sdk/pull/241

Reply:
Addressed with https://github.com/yearn/yearn-sdk/pull/234/commits/92af5702ac5920bfa3036aa97daf60f85e70ef0f